addons/web/static/src/js/env.js

```diff
@@ -0,0 +1,92 @@
+odoo.define("web.env", function (require) {
+"use strict";
+
+const ajax = require('web.ajax');
+const config = require("web.config");
+const { _lt, _t, bus } = require("web.core");
+const rpc = require("web.rpc");
+const session = require("web.session");
+const utils = require("web.utils");
+
+const qweb = new owl.QWeb({ translateFn: _t });
+
+function ajaxJsonRPC() {
+    return ajax.jsonRpc(...arguments);
+}
+
+function getCookie() {
+    return utils.get_cookie(...arguments);
+}
+
+function httpRequest(route, params={}, readMethod='json') {
+    const formData = new FormData();
+    for (const key in params) {
+        if (key === 'method') {
+            continue;
+        }
+        const value = params[key];
+        if (Array.isArray(value) && value.length) {
+            for (const val of value) {
+                formData.append(key, val);
+            }
+        } else {
+            formData.append(key, value);
+        }
+    }
+
+    return fetch(route, {
+        method: params.method || 'POST',
+        body: formData,
+    }).then(response => response[readMethod]());
+}
+
+function navigate(url, params) {
+    window.location = $.param.querystring(url, params);
+}
+
+function blockUI(params) {
+    $.blockUI(params);
+}
+
+function unblockUI() {
+    $.unblockUI();
+}
+
+function performRPC(params, options) {
+    const query = rpc.buildQuery(params);
+    return session.rpc(query.route, query.params, options);
+}
+
+function reloadPage() {
+    window.location.reload();
+}
+
+function setCookie() {
+    utils.set_cookie(...arguments);
+}
+
+// There should be as much dependencies as possible in the env object.
+// This will allow an easier testing of components.
+// See https://github.com/odoo/owl/blob/master/doc/reference/environment.md#content-of-an-environment
+// for more information on environments.
+return {
+    _lt,
+    _t,
+    bus,
+    config,
+    qweb,
+    rpc: performRPC,
+    services: {
+        ajaxJsonRPC,
```

Comment:
rpc methods should use (un)blockui fn from env (so they need to be more elaborate, see misc.js)